### PR TITLE
ENS compatibility

### PR DESCRIPTION
**Pull request by James-Sangalli**
For #810 

Allows you to write/paste any valid ENS name into the box and it will be translated into the corresponding address

### CODE REVIEW

File: AlphaWallet/Tokens/Coordinators/GetENSOwnerCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    ) {
+
+        //if already an address, send back the address
+        if let ethAddress = EthereumAddress(input) {
+            completion(.success(ethAddress))
+            return
+        }
+
+        //if it does not contain .eth, then it is not a valid ens name
+        if !input.contains(".eth") {
```
Comment (hboon) [inline AlphaWallet/Tokens/Coordinators/GetENSOwnerCoordinator.swift]: We can use `hasSuffix(".eth")` instead. But it appears that there's also .xyz and .luxe. Maybe we can't perform this check reliably?

File: AlphaWallet/Tokens/Coordinators/GetENSOwnerCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        guard let promise = contractInstance.method(function.name, parameters: [node] as [AnyObject], options: nil) else {
+            completion(.failure(AnyError(Web3Error(description: "Error calling \(function.name)() on \(contractAddress.address)"))))
+            return
+        }
+
+        promise.callPromise(options: nil).done { result in
+            //if null address is returned (as 0) we count it as invalid
+            //this is because it is not assigned to an ENS and puts the user in danger of sending funds to null
+            if let owner = result["0"] as? EthereumAddress {
+                completion(.success(owner))
```
Comment (hboon) [inline AlphaWallet/Tokens/Coordinators/GetENSOwnerCoordinator.swift]: If it fails to resolve, the textbox displays "0x0000000000000000000000000000000000000000". Maybe we can hardcode a check for 0x0 and don't display it?

**Comment by James-Sangalli** [comment]
@hboon ok.

Actually, it is better to simply check for "." as any text input that has a dot in it can be implied to be an ENS name, it is also compatible with subdomains. User is protected because an invalid name will still return a null value.

**Comment by James-Sangalli** [comment]
@hboon 2 is not an issue, it is able to detect the ens name if you use the black and white paste button

**Comment by hboon** [comment]
@James-Sangalli I was referring to this white on black Paste button. Does it trigger ENS resolution?

simulator screen shot - iphone 6s - 2018-11-10 at 22 04 34

**Comment by hboon** [comment]
Another future enhancement when we get the Ethereum name and address to show up together: handle when user types in an ENS name, maybe start resolving say, 0.5 seconds after user stops typing.

**Pull request merged**